MeshLib/IO/MPI_IO/NodePartitionedMeshReader.cpp

```diff
@@ -184,8 +184,121 @@ MeshLib::NodePartitionedMesh* NodePartitionedMeshReader::readBinary(
     setElements(mesh_nodes, ghost_elem_data, mesh_elems, process_ghost);
 
     //----------------------------------------------------------------------------------
-    return newMesh(BaseLib::extractBaseName(file_name_base),
-               mesh_nodes, glb_node_ids, mesh_elems);
+    // read the properties
+    MeshLib::Properties p(readPropertiesBinary(file_name_base));
+
+    return newMesh(BaseLib::extractBaseName(file_name_base), mesh_nodes,
+                   glb_node_ids, mesh_elems, p);
+}
+
+MeshLib::Properties NodePartitionedMeshReader::readPropertiesBinary(
+    const std::string& file_name_base) const
+{
+    const std::string fname_cfg = file_name_base +
+                                  "_partitioned_properties_cfg" +
+                                  std::to_string(_mpi_comm_size) + ".bin";
+    std::ifstream is(fname_cfg.c_str(), std::ios::binary | std::ios::in);
+    if (!is)
+    {
+        WARN("Could not open file '%s' in binary mode.", fname_cfg.c_str());
+        return MeshLib::Properties();
+    }
+    std::size_t number_of_properties = 0;
+    is.read(reinterpret_cast<char*>(&number_of_properties), sizeof(std::size_t));
+    std::vector<boost::optional<MeshLib::IO::PropertyVectorMetaData>> vec_pvmd(
+        number_of_properties);
+    for (std::size_t i(0); i < number_of_properties; ++i)
+    {
+        vec_pvmd[i] = MeshLib::IO::readPropertyVectorMetaData(is);
+        if (!vec_pvmd[i])
+        {
+            OGS_FATAL(
+                "Error in NodePartitionedMeshReader::readPropertiesBinary: "
+                "Could not read the meta data for the PropertyVector %d",
+                i);
+        }
+    }
+    for (std::size_t i(0); i < number_of_properties; ++i)
+    {
+        DBUG("[%d] +++++++++++++", _mpi_rank);
+        MeshLib::IO::writePropertyVectorMetaData(*(vec_pvmd[i]));
+        DBUG("[%d] +++++++++++++", _mpi_rank);
+    }
+    auto pos = is.tellg();
+    auto offset =
+        pos +
+        static_cast<long>(_mpi_rank *
+                          sizeof(MeshLib::IO::PropertyVectorPartitionMetaData));
+    is.seekg(offset);
+    boost::optional<MeshLib::IO::PropertyVectorPartitionMetaData> pvpmd(
+        MeshLib::IO::readPropertyVectorPartitionMetaData(is));
+    if (!pvpmd)
```

Comment:
Suggest to replace it with
```
 boost::optional<MeshLib::IO::PropertyVectorPartitionMetaData> pvpmd_i(
        MeshLib::IO::readPropertyVectorPartitionMetaData(is));
MPI_C_BOOL pvpmd_read_ok_i = static_cast<MPI_C_BOOL>(pvpmd_i);
MPI_C_BOOL pvpmd_read_ok; 
MPI_Allreduce( &pvpmd_read_ok_i, &pvpmd_read_ok, 1, MPI_C_BOOL, MPI_LOR, MPI_COMM_WORLD);
if (!pvpmd_read_ok)
```
Otherwise, one rank fails and quit while some other rank may be good and still run.